### PR TITLE
Made option to always load the scripts over https.

### PR DESCRIPTION
**Pull request by rvdgraaf**
# Describe your changes

Added option to always load the external scripts of a template over https.

## Type of change

Please check only ONE option.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How was this tested?

Scripts where served over http on a page running on http. After this change and adjusting the setting, scripts were served over https.

# Checklist before requesting a review
- [X] I have reviewed and tested my changes
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I selected `develop` as the base branch and not `main`, or the pull request is a hotfix that needs to be done directly on `main`
- [X] I double checked all my changes and they contain no temporary test code, no code that is commented out and no changes that are not part of this branch
- [X] I added new unit tests for my changes if applicable

# Related pull requests

N/A 

# Link to Asana ticket

N/A

### CODE REVIEW

**Review by GilianJuice — CHANGES_REQUESTED** [review]
I'm sorry but I don't understand the use case of this change. You changed a function that converts a relative URI to an absolute URI. Why would you ever want to have a website run on http, but load it's scripts, of the same domain, via https? That doesn't make any sense. Just run the entire website on https then. Besides that, don't like it if you mix http and https on the same domain, everything should be the same, not mixed. I think this change might cause more problems than it solves.

So please explain why you made this change and what the use case is exactly. At the moment I don't understand the point of this change and don't want to merge something that doesn't seem to have a good use.

File: GeeksCoreLibrary/Modules/Templates/Services/PagesService.cs [hunk 1]
```diff
@@ -472,7 +472,7 @@ public async Task<PageViewModel> CreatePageViewModelAsync(List<PageResourceModel
             foreach (var externalScript in externalScripts.Where(externalScript => !externalScript.Uri.IsAbsoluteUri))
             {
                 // Turn relative URI into absolute.
-                externalScript.Uri = new Uri($"{HttpContextHelpers.GetBaseUri(httpContextAccessor?.HttpContext)}{externalScript.Uri.OriginalString}");
+                externalScript.Uri = new Uri($"{HttpContextHelpers.GetBaseUri(httpContextAccessor?.HttpContext, (await objectsService.FindSystemObjectByDomainNameAsync("always_use_https", "false")).Equals("true", StringComparison.OrdinalIgnoreCase))}{externalScript.Uri.OriginalString.TrimStart('/')}");
```
[the remark this answers is not in the document]
Comment (rvdgraaf) [reply GeeksCoreLibrary/Modules/Templates/Services/PagesService.cs]: Removed the setting, so the name of the setting is not relevant anymore.

**Comment by rvdgraaf** [comment]
The case is that we have a site that runs behind a load balancer. The internal communication between the load balancer and the web server is via HTTP. Without this change the GCL builds the URLs of the external scripts with http. The site does run on https. Without this adjustment you will get a 'Mixed content' error in your browser and the scripts will not be loaded.

I have applied the changes as discussed.

**Review by GilianJuice — CHANGES_REQUESTED** [review]
Sorry, but I think you changed too much now. The extra parameter you added to `GetBaseUri` was fine. We can just set it hard-coded to true in some/most places that use this method. Or make the default value true and set it to false where we don't want it. Because this method is also used in logging, we don't want to force https there, that could become very confusing. In logging we want to know the actual URL that was used.

**Comment by rvdgraaf** [comment]
Applied changes

**Review by GilianJuice — APPROVED** [review]

**Pull request merged**